src/app/home-page/recent-events/recent-events.component.html

```diff
@@ -1,103 +1,189 @@
-            <mat-divider *ngIf="!last" class="custom-margin"></mat-divider>
+            edited the collection
+            <a [routerLink]="event | recentEvents: 'collectionLink'">{{ event.collection?.name }}</a>
           </div>
-          <div *ngSwitchCase="EventType.UNPUBLISHENTRY">
-            <strong>{{ event.user.username }} </strong> unpublished the {{ event?.tool || event?.apptool ? 'tool' : 'workflow' }}
-            <a
-              [routerLink]="
-                event?.apptool
-                  ? '/tools/' + event?.apptool?.full_workflow_path
-                  : event?.tool
-                  ? '/tools/' + event?.tool?.tool_path
-                  : '/workflows/' + event?.workflow?.full_workflow_path
-              "
-              >{{
-                (event?.tool | entryToDisplayName) || (event?.workflow | entryToDisplayName) || (event?.apptool | entryToDisplayName)
-              }}</a
+          <div *ngSwitchCase="EventType.REMOVEFROMCOLLECTION" class="truncate-text-2">
+            <strong
+              ><a class="no-underline" [routerLink]="'/users/' + event.initiatorUser?.username">{{
+                event.initiatorUser?.username
+              }}</a></strong
             >
-            <div class="subtitle">{{ 'on ' + (event.dbCreateDate | date: 'medium') }}</div>
-            <mat-divider *ngIf="!last" class="custom-margin"></mat-divider>
+            removed the {{ event | recentEvents: 'entryType' }}
+            <a [routerLink]="event | recentEvents: 'entryLink'">{{ event | recentEvents: 'displayName' }}</a>
+            from the collection
+            <a [routerLink]="event | recentEvents: 'collectionLink'">{{ event.collection?.name }}</a>
           </div>
+          <div class="subtitle">{{ event.dbCreateDate | date: 'medium' }}</div>
         </div>
       </div>
+      <div
+        *ngIf="
+          event.type === EventType.ADDVERSIONTOENTRY || event.type === EventType.PUBLISHENTRY || event.type === EventType.UNPUBLISHENTRY;
+          else orgImg
+        "
+      >
+        <a *ngIf="event.type !== EventType.UNPUBLISHENTRY; else noLink" [routerLink]="event | recentEvents: 'entryLink'">
+          <img
+            src="../../../assets/svg/sub-nav/{{ event | recentEvents: 'entryType' }}.svg"
+            class="site-icons-medium mx-3"
+            alt="{{ event | recentEvents: 'entryType' }} icon"
+          />
+        </a>
+        <!-- Do not link to an entry that doesn't exist - 404 -->
+        <ng-template #noLink>
+          <img
+            src="../../../assets/svg/sub-nav/{{ event | recentEvents: 'entryType' }}.svg"
+            class="site-icons-medium mx-3"
+            alt="{{ event | recentEvents: 'entryType' }} icon"
+          />
+        </ng-template>
+      </div>
+      <ng-template #orgImg>
+        <a *ngIf="event.organization?.avatarUrl; else orgIcon" [routerLink]="event | recentEvents: 'orgLink'">
```

Comment:
One failure mode is when the organization specifies an `avatarUrl`, but the image pointed at by the URL doesn't exist or is not valid/loadable.  In that case, to prevent a "broken" icon from being displayed, we implemented some fallback code to load a default image.  We should probably include it here.  See the `appFallback` directive at this line: https://github.com/dockstore/dockstore-ui2/blob/d4f5ccb4c3c1c9ac4be2f4cf72e40f7a33277948/src/app/organizations/organizations/organizations.component.html#L69

Reply:
Works great, thanks Steve :)